src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -612,3 +635,22 @@ def _generate_output_path(self, config: Optional[Configuration]) -> Path:
                 )
                 logging.getLogger(LOGGER_NAME).warning(warning_message)
         return (path / str(self.name)).resolve()
+
+    @classmethod
+    def _load_from_source(cls, source: Union[str, Path], params_override=None, **kwargs) -> "Run":
+        """Load run from run record source folder."""
+        source = Path(source)
+        params_override = params_override or {}
+        if not source.exists() or not source.is_dir():
+            raise UserErrorException(f"Invalid run source: {source!r}. Expecting a valid run source folder.")
+
+        return cls(
+            name=source.name,
+            source=source,
+            run_source=RunInfoSources.EXISTING_RUN,
+            status=RunStatus.COMPLETED,  # failed run shall not be created in this way
```

Comment:
if we don't expect failed run here, shall we add some status check at the very beginning of this function?

Reply:
ah, maybe it's kind of hard to do at the top as you may not know it at all, but somehow I think we might need a check?

Reply:
It's considered, but the problem here is we don't know the status of this run, this run folder can be downloaded or simply shared. For the download case, maybe we can write some additional info in the run metadata like start/end time, for the later case,there is no good solution yet.